odoo/addons/base/models/ir_actions_report.py

```diff
@@ -402,6 +403,7 @@ def _run_wkhtmltopdf(
         document.
 
         :param list[str] bodies: The html bodies of the report, one per page.
+        :param report_ref: report reference that is needed to get report paperformat.
```

Comment:
This could be better named and documented: what is `report_ref` supposed to be and what's the type of that parameter? You could refer people to the docstring of `get_report()` perhaps.

Anyway, why is it mandatory? As the docstring says, `_run_wkhtmltopdf()` is a helper that transforms HTML input into a PDF. It is valid to call it without a specific report, and in that case, the default of using the current company's paperformat is fine, no?

If callers are required to pass an explicit report, it doesn't make sense for this method to be `@api.model`. Do we want to change that, and require callers to lookup the report first? I don't think so.

Reply:
Indeed, I will make report_ref optional again add the conditional statement for calling get_paperformat() function for paperformat_id.